pytest/tests/loadtest/locust/locustfiles/sweat.py

```diff
@@ -54,9 +58,38 @@ def record_batch(self):
             ])
         self.send_tx(tx, locust_name="Sweat record batch")
 
+    @tag("storage-stress-test")
+    @task
+    def record_batch_of_large_batches(self):
+        # ensure large enough state by creating more sweat users
+        while len(self.sweat.registered_users) < 1000:
+            SweatUser(self.environment).on_start()
+
+        rng = random.Random()
+        # just around 300Tgas
+        batch_size = rng.randint(700, 750)
+        receivers = self.sweat.random_receivers(self.account_id, batch_size)
+        tx = SweatMintBatch(
+            self.sweat.account.key.account_id,
+            self.sweat.oracle,
+            [[account_id, rng.randint(1000, 3000)] for account_id in receivers])
+        self.send_tx(tx, locust_name="Sweat record batch (stress test)")
+
     def on_start(self):
         super().on_start()
-        self.sweat = self.environment.sweat
+        # We have one oracle account per worker. Sharing a single access key
+        # means potential conflicts in nonces when we mint new tokens through
+        # batches. Hence, let's add a new access key to the oracle account for
+        # each sweat user.
+        oracle = self.environment.sweat.oracle
+        user_orcale_key = key.Key.from_random(oracle.key.account_id)
+        self.send_tx_retry(AddFullAccessKey(oracle, user_orcale_key),
+                           "add user key to oracle")
+
+        self.sweat = copy.copy(self.environment.sweat)
```

Comment:
Yeah this is non-obvious, at the very least it will need a comment.

So what's happening here is that I want to reference the same contract but use a different key to access it.

It's hacky like this because `class SweatContract` owns both the access key and also the list of `registered_users`.
The access key must NOT be shared, `registered_users` must be shared among all users.

Most other things I don't care about whether they are shared or not. But these other fields are read-only, so sharing seems more efficient. Hence a shallow copy of `sweat` seemed like the easiest way to implement it. At least as I understand it, it let's me overwrite `sweat.oracle` without affecting the copy used by other users. But it will share the `registered_users` list as well as the code field etc.

Creating a new constructor was my first approach but it looks bad IMO to do this because it directly accesses fields inside the class:

```python
sweat_copy = SweatContract(sweat.account, sweat.oracle, sweat.code)
``` 

where `account` and `code` are fields of `FTContract` and `oracle` is a field of `SweatContract(FTContract)`. 
(God I hate OOP/inheritance, especially in dynamically typed languages... oh well, I wrote the code this way :roll_eyes:)

I don't know, do you think this would be better than a shallow copy and overwriting a field?

Or would a dedicated clone-like method along these lines make sense to you?

```python
# in the contract class
def use_oracle_access_key(self, oracle_key: key.Key, node: NearNodeProxy) -> SweatContract:
   copied = copy.copy(self)
   copied.oracle = Account(oracle_key)
   self.oracle.refresh_nonce(node.node)
   return copied

# in the user class
self.sweat = self.environment.sweat.use_oracle_access_key(user_oracle_key)
```


It looks a bit over-engineered to me but no strong opinion.

Reply:
Hm, actually 

```python
sweat_copy = SweatContract(sweat.account, sweat.oracle, sweat.code)
```

doesn't work because it doesn't share the registered_users

Reply:
> So what's happening here is that I want to reference the same contract but use a different key to access it.

I see! Reading through the code, I don't quite understand why overwriting `sweat.oracle` is necessary for the desired effect of using a different access key. I see three places where the oracle field is used:
- [When the SweatContract is constructed](https://github.com/near/nearcore/blob/f28033b745d825d9e3ecbaaa1b6ce59127d33c86/pytest/tests/loadtest/locust/common/sweat.py#L21) to set `ft_distributor`, but overwriting this field after constructor won't influence `ft_distributor` 
- [When contract is initialized](https://github.com/near/nearcore/blob/f28033b745d825d9e3ecbaaa1b6ce59127d33c86/pytest/tests/loadtest/locust/common/sweat.py#L32C14-L32C29), but that is called [only once on the MasterRunner](https://github.com/near/nearcore/blob/f28033b745d825d9e3ecbaaa1b6ce59127d33c86/pytest/tests/loadtest/locust/common/sweat.py#L155), so no effect here?
- [`record_batch_of_large_batches` added in this PR](https://github.com/near/nearcore/pull/9385/files#diff-dd0c2c70c41c997fd0cb86472624ed02f913c6b3ba0e4a1a26b18d429449ee34R73), but here we manually extract `oracle` from `self.sweat` - we might as well introduce a field `self.oracle` in `SweatUser` and store it there directly

Reply:
> we might as well introduce a field self.oracle in SweatUser and store it there directly

Yes, that actually makes a lot of sense :)

Reply:
I think there is potential to simplify some of the logic around oracles in `SweatContract`. But I will merge it now with the minimal change to apply your suggestion (use `self.oracle` inside `SweatUser`) because the storage team is waiting for this to be in master.